### PR TITLE
(maint) Add augeas 1.0.0 module

### PR DESCRIPTION
**Pull request by joshcooper**

### CODE REVIEW

**Comment by melissa** [comment]
from puppet-agent.x86_64 0:5.5.3.290.g17d9740-1.el7:
```
[root@v047kcmjtbimv73 ~]# tree /opt/puppetlabs/puppet/vendor_modules/augeas_core/
/opt/puppetlabs/puppet/vendor_modules/augeas_core/
├── appveyor.yml
├── CHANGELOG.md
├── Gemfile
├── lib
│   └── puppet
│       ├── feature
│       │   └── augeas.rb
│       ├── provider
│       │   └── augeas
│       │       └── augeas.rb
│       └── type
│           └── augeas.rb
├── LICENSE
├── metadata.json
├── Rakefile
├── README.md
├── REFERENCE.md
└── spec
    ├── acceptance
    │   ├── nodesets
    │   │   └── default.yml
    │   └── tests
    │       ├── hosts_spec.rb
    │       ├── puppet_spec.rb
    │       └── services_spec.rb
    ├── default_facts.yml
    ├── fixtures
    │   └── unit
    │       └── provider
    │           └── augeas
    │               └── augeas
    │                   ├── augeas
    │                   │   └── lenses
    │                   │       └── test.aug
    │                   ├── etc
    │                   │   ├── fstab
    │                   │   ├── hosts
    │                   │   └── test
    │                   └── test.aug
    ├── spec_helper_acceptance.rb
    ├── spec_helper.rb
    └── unit
        ├── provider
        │   └── augeas
        │       └── augeas_spec.rb
        └── type
            └── augeas_spec.rb

22 directories, 25 files
```

File: configs/components/module-puppetlabs-augeas_core.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"url":"git@github.com:puppetlabs/puppetlabs-augeas_core.git","ref":"refs/tags/1.0.0"}
```
Comment (caseywilliams) [inline configs/components/module-puppetlabs-augeas_core.json]: Other components tend to use the read-only `git://` protocol -- I don't really see a strong reason _not_ to use `git@` here, though.

Comment (joshcooper) [reply configs/components/module-puppetlabs-augeas_core.json]: ah, my bad. The repo was private, let me fix that

**Comment by joshcooper** [comment]
jenkins please test this

**Comment by joshcooper** [comment]
@caseywilliams I think this is ready now. I rebuilt the module with the `git://` ref, and get the same result:

```
[root@mheo7hzuk5t3yg6 ~]# tree /opt/puppetlabs/puppet/vendor_modules/augeas_core/
/opt/puppetlabs/puppet/vendor_modules/augeas_core/
├── appveyor.yml
├── CHANGELOG.md
├── Gemfile
├── lib
│   └── puppet
│       ├── feature
│       │   └── augeas.rb
│       ├── provider
│       │   └── augeas
│       │       └── augeas.rb
│       └── type
│           └── augeas.rb
├── LICENSE
├── metadata.json
├── Rakefile
├── README.md
├── REFERENCE.md
└── spec
    ├── acceptance
    │   ├── nodesets
    │   │   └── default.yml
    │   └── tests
    │       ├── hosts_spec.rb
    │       ├── puppet_spec.rb
    │       └── services_spec.rb
    ├── default_facts.yml
    ├── fixtures
    │   └── unit
    │       └── provider
    │           └── augeas
    │               └── augeas
    │                   ├── augeas
    │                   │   └── lenses
    │                   │       └── test.aug
    │                   ├── etc
    │                   │   ├── fstab
    │                   │   ├── hosts
    │                   │   └── test
    │                   └── test.aug
    ├── spec_helper_acceptance.rb
    ├── spec_helper.rb
    └── unit
        ├── provider
        │   └── augeas
        │       └── augeas_spec.rb
        └── type
            └── augeas_spec.rb

22 directories, 25 files
```

**Pull request merged**